src/libraries/System.Runtime.InteropServices/gen/ComInterfaceGenerator/GeneratedComInterfaceAttributeAnalyzer.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.InteropServices;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+#nullable enable
+namespace Microsoft.Interop
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class GeneratedComInterfaceAttributeAnalyzer : DiagnosticAnalyzer
+    {
+        private const string _generatedComInterfaceAttributeName = "GeneratedComInterfaceAttribute";
+        private const string _generatedComInterfaceAttributeNamespace = "System.Runtime.InteropServices.Marshalling";
+#if DEBUG
+        private const string _generatedComInterfaceAttributeAssemblyName = "Microsoft.Interop.Ancillary";
+#else
+        private const string _generatedComInterfaceAttributeAssemblyName = "System.Runtime.InteropServices";
+#endif
+        private const string _interfaceTypeAttributeName = "InterfaceTypeAttribute";
+        private const string _interfaceTypeAttributeNamespace = "System.Runtime.InteropServices";
+        private const string _interfaceTypeAttributeAssemblyName = "System.Runtime.InteropServices";
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; }
+            = ImmutableArray.Create(GeneratorDiagnostics.InterfaceTypeNotSupported);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+            if (!Debugger.IsAttached)
+                context.EnableConcurrentExecution();
```

Comment:
We should always enable concurrent execution for the analyzer. If that makes the development experience worse, we can update the test harness to disable concurrent execution for all of our analyzer tests when the debugger is attached (which wouldn't be that bad of an idea).